bundles/core/src/main/java/com/adobe/cq/wcm/core/components/services/link/LinkProcessor.java

```diff
@@ -0,0 +1,33 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2021 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.services.link;
+
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A service that can process a given link url
+ *
+ * @since com.adobe.cq.wcm.core.components.services.link 1.0.0
+ */
+public interface LinkProcessor {
+
+    /**
+     * Process a given link
+     * @param linkUrl the link url
+     * @return the processed link if the processor can handle the link, otherwise null
+     */
+    @Nullable String process(@Nullable String linkUrl);
+}
```

Comment:
we may want to introduce a method that verifies if the processor can process the given URL. E.g:
```
boolean canProcess(@Nullable String linkUrl);
```
This would enable defining which kind of links the processor can process based on regex in OSGI configuration.